ENIGMAsystem/SHELL/Audio_Systems/OpenAL/ALsystem.cpp

```diff
@@ -154,11 +149,6 @@ namespace enigma {
 
   int sound_add_from_stream(int id, size_t (*callback)(void *userdata, void *buffer, size_t size), void (*seek)(void *userdata, float position), void (*cleanup)(void *userdata), void *userdata)
```

Comment:
Id was always the sound resource id where the sound should go. I really don't get why it was structured this way. This is why I suggested renaming the function to `insert` or `replace` or something, because sorlok added the check because he was afraid that `sound_add` was reusing ids when it was really `sound_replace` reusing them.

I'm not going to pass SoundResource* because if I do that then this function will get bigger. `sound_add` and `sound_replace` are smaller because of the way this is set up. I will simply focus my efforts on #1125 and then start with DirectSound and OpenAL as the first systems to implement the new universal resource storage container.

Reply:
Scratch that, if you read my comments/changes now after I looked at `alureBufferDataFromFile` it is actually a lot more logical now and even less lines of code! So now it's basically the way you want it. I am glad I looked into all this because it's definitely improving things.